src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperAttributeList.cs

```diff
@@ -51,124 +50,77 @@ public TagHelperAttributeList(IEnumerable<TagHelperAttribute> attributes)
                     throw new ArgumentNullException(nameof(value));
                 }
 
-                if (value.Name == null)
-                {
-                    throw new ArgumentException(
-                        Resources.FormatTagHelperAttributeList_CannotAddWithNullName(
-                            typeof(TagHelperAttribute).FullName,
-                            nameof(TagHelperAttribute.Name)),
-                        nameof(value));
-                }
-
-                Attributes[index] = value;
+                Items[index] = value;
             }
         }
 
         /// <summary>
-        /// Gets the first <see cref="TagHelperAttribute"/> with <see cref="TagHelperAttribute.Name"/> matching
-        /// <paramref name="name"/>. When setting, replaces the first matching
-        /// <see cref="TagHelperAttribute"/> with the specified <paramref name="value"/> and removes any additional
-        /// matching <see cref="TagHelperAttribute"/>s. If a matching <see cref="TagHelperAttribute"/> is not found,
-        /// adds the specified <paramref name="value"/> to the end of the collection.
-        /// </summary>
+        /// Replaces the first <see cref="TagHelperAttribute"/> whose <see cref="TagHelperAttribute.Name"/> matches the
+        /// specified <paramref name="name"/> and removes any additional matching <see cref="TagHelperAttribute"/>s. If
```

Comment:
Stick w/ `with <see cref="TagHelperAttribute.Name"/> matching <paramref name="name"/>` i.e. `Replaces the first <see cref="TagHelperAttribute"/> with ...`.